File: docs/tutorial/python/t7_search_author.py
import sys
import flax.searchclient

# open a connection to Flax Server
conn = flax.searchclient.Client('http://localhost:8080')

# get a reference to the database
db = conn.db('books')

# do the search and get the results
results = db.search_structured(filter=['author:%s' % sys.argv[1]])

# print a summary of the results
print '%d to %d of %s results' % (results.start_rank + 1, 
    results.end_rank,
    results.matches_human_readable_estimate)

# print the rank and title of each result
for hit in results.results:
    print '%s. %s' % (hit.rank + 1, hit.data['title'][0])

File: flax_search_service/docs/tutorial/perl/t7_search_author.pl
#!/usr/bin/perl

use strict;
use warnings;
use Error qw(:try);

use Flax;
use Flax::Error;
use Flax::Field;
use Flax::SummarySettings;

my $flax = Flax->new('url'=>'http://localhost:8080');
my $db = $flax->getDatabase('books');

# do the search and get the results
my $res = $db->searchStructured('', '', '', '', [['author', $ARGV[0] ]]);

if ($res->end_rank)
{
    # print a summary of the results
    print $res->start_rank + 1 ." to ". $res->end_rank ." of ".
        $res->matches_human_readable_estimate . " results:\n";
    
    foreach (@{$res->results})
    {
        print "(". ($_->rank + 1) .") ". $_->data->{'title'}->[0] ."\n";
    }
}
else
{
    print "no results\n";
}